saleor/graphql/product/resolvers.py

```diff
@@ -25,16 +25,22 @@ def resolve_categories(info, level=None, **_kwargs):
     return qs.distinct()
 
 
-def resolve_collection_by_slug(info, slug):
+def resolve_collection_by_slug(info, slug, channel_slug):
     requestor = get_user_or_app_from_context(info.context)
     return (
-        models.Collection.objects.visible_to_user(requestor).filter(slug=slug).first()
+        models.Collection.objects.visible_to_user_with_channel_slug(
```

Comment:
In ProductsQueryset we use `visible_to_user` and it has a channel as param. 
We are not able to return `visible_to_user` for non-staff users without passing the channel as param.

Reply:
The changes are already implemented, I didn't commit changes because of extend this PR with removing homepage collection.